spec/features/account_connected_apps_spec.rb

```diff
@@ -78,27 +85,47 @@
     expect(page).to_not have_content(identity.display_name)
   end
 
-  scenario 'changing email shared with SP' do
+  scenario 'changing email shared with SP', :js do
     within('li', text: identity.display_name) do
       expect(page).to have_content(t('account.connected_apps.email_not_selected'))
       click_link(t('help_text.requested_attributes.change_email_link'))
     end
 
-    expect(page).to have_field(user.email) { |field| !field[:checked] }
+    expect(page).not_to have_checked_field(user.email)
```

Comment:
Huh, TIL about `have_checked_field`, that's handy!

Reply:
Interestingly I think I'm noticing that the `validates` adds the required attribute. I'm not sure why Capybara isn't finding the radio button. I'm still working that out. 
I'm also noticing in manual testing that it does get focused. Also, probably not related but the tab-order is a little unexpected. You can't seem to tab to the second radio button - it goes right to the Select button.

Reply:
Those inputs were hidden using CSS so Capybara wasn't finding them. I was able to get the test working by telling it to look for `visible: false` on the selectors